src/main/java/org/opensearch/security/dlic/rest/api/InternalUsersApiAction.java

```diff
@@ -50,17 +52,19 @@
 
 public class InternalUsersApiAction extends PatchableResourceApiAction {
     static final List<String> RESTRICTED_FROM_USERNAME = ImmutableList.of(
-        ":" // Not allowed in basic auth, see https://stackoverflow.com/a/33391003/533057
+            ":" // Not allowed in basic auth, see https://stackoverflow.com/a/33391003/533057
     );
 
     private static final List<Route> routes = addRoutesPrefix(ImmutableList.of(
             new Route(Method.GET, "/user/{name}"),
             new Route(Method.GET, "/user/"),
+            new Route(Method.GET, "/user/{name}/authtoken"),
```

Comment:
Shouldn't be a `GET`, I think this should be a `POST` since its modifying the user and it isn't going to generate the same result if called multiple times.  Used https://www.baeldung.com/rest-http-put-vs-post to look into this, I'd be curious what you think.

Reply:
Yeah that is a good point. I did not think about the distinction because it feels like you are "getting" something. I see that it should be the same result each time though, and this may not be the case.

Reply:
The result doesn't have to be the same each time, I think it makes sense for their to be different responses in this case. Typically 'PUT' calls do return a consistent result that is all

Reply:
Yeah, sorry, I should have been clearer. I meant that I see that `GET` should give the same response each time and since this won't I should use a different request type.